frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/backgroundables/SlideOutAndDeleteHandler.java

```diff
@@ -0,0 +1,137 @@
+package com.virtuslab.gitmachete.frontend.actions.backgroundables;
+
+import static com.virtuslab.gitmachete.frontend.resourcebundles.GitMacheteBundle.getString;
+
+import java.util.Collections;
+
+import com.intellij.openapi.application.ModalityState;
+import com.intellij.openapi.progress.ProgressIndicator;
+import com.intellij.openapi.progress.Task;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.vcs.VcsException;
+import com.intellij.openapi.vcs.VcsNotifier;
+import com.intellij.openapi.vfs.VirtualFile;
+import com.intellij.util.Consumer;
+import com.intellij.util.ModalityUiUtil;
+import git4idea.branch.GitBrancher;
+import git4idea.config.GitConfigUtil;
+import git4idea.repo.GitRepository;
+import lombok.CustomLog;
+import lombok.experimental.ExtensionMethod;
+import lombok.val;
+import org.checkerframework.checker.guieffect.qual.UIEffect;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+import com.virtuslab.gitmachete.frontend.actions.dialogs.DeleteBranchOnSlideOutSuggestionDialog;
+import com.virtuslab.gitmachete.frontend.resourcebundles.GitMacheteBundle;
+import com.virtuslab.gitmachete.frontend.ui.api.table.BaseEnhancedGraphTable;
+import com.virtuslab.qual.guieffect.UIThreadUnsafe;
+
+@ExtensionMethod(GitMacheteBundle.class)
+@CustomLog
+public class SlideOutAndDeleteHandler {
```

Comment:
the main reason for extracting this class is that this is a scope where `gitRepository` is not null

Reply:
I'm assuming there's :zero: changes in logic here 😅